base/common/src/main/java/org/dogtagpki/common/CAInfo.java

```diff
@@ -93,10 +125,96 @@ public boolean equals(Object obj) {
         if (keyWrapAlgorithm == null) {
             if (other.keyWrapAlgorithm != null)
                 return false;
-        } else if (!keyWrapAlgorithm.equals(other.keyWrapAlgorithm))
+        } else if (!keyWrapAlgorithm.equals(other.keyWrapAlgorithm)) {
+            return false;
+        } else if (!rsaPublicKeyWrapAlgorithm.equals(other.rsaPublicKeyWrapAlgorithm)) {
+            return false;
+        } else if (!caRsaPublicKeyWrapAlgorithm.equals(other.caRsaPublicKeyWrapAlgorithm)) {
             return false;
+        }
+
         return true;
     }
 
+
+    @Override
+    public Element toDOM(Document document) {
```

Comment:
@edewata this is the porting of the [PR#4162](https://github.com/dogtagpki/pki/pull/4162)  where Chris raises the same problem. It can be removed without issue (I think) but it is there because the superclass implements the methods and if caller invoke the endpoint get a wrong xml. When the superclass drop the *DOM methods we can drops in all the extending classes. This was the outcome of that discussion. If you think it is better to drop them in any case we can work on that.

Reply:
To my understanding the `CAInfo` and `KRAInfo` are existing classes, so the XML mappings for them are already broken. Since there seems to be no complaints about them so far, I'd say anybody that needs them is using JSON, so there's no need to fix the XML mappings for these classes.